### ISSUE TITLE
Game freezes

### ISSUE THREAD

**Issue by luk3ZPL** [Return-To-The-Roots/s25client#863]
Which version is affected?
s25rttr_20180813

What did you do/shall be done to reproduce the bug?
Check my save [map created by random map generator]
http://www.mediafire.com/file/7etn1g9x1pjmlx4/bug_%2526_screenshot.7z/file

What is expected? (Unless it is a crash)
-

What happens instead?
-

**Comment by Flamefire** [comment]
Thanks for the report, but please split unrelated issues next time. That makes solving them easier.

I found the 2nd bug and fixed it (hopefully, it's not really easy...)

About the freeze: No idea. Wasn't able to reproduce this at all (Win10, minimize, pause/unpause multiple times etc.) Are you using SDL (video settings)? Any kind of restricted frame rate? How often does it happen? On other maps to?

**Comment by luk3ZPL** [comment]
My save freeze after 2-3 sec after start (currently I don't doing anything just load a save). I have log file:
Yes, I'm using RTTR x64, SDL and limited frame rate to 60 fps. This happen in later game but maybe 3 times.
I play only on random maps, so I don't know whether standard maps are affected or not.
In the screenshot you can also see bug with AI road when I captured their watchtower (probably road upgraded during capturing).
I also noticed when autosave save the game in the same moment while I got a message in game then the game 
create file which hasn't any name just (.sav).

log:
(...)
RttR crashed. Backtrace:
0x41492d
0x143cd40
0x143ce38
0x143d5b0
0x143e2c0
0x143c760
Connection to 5.9.167.129:4123
Successfully connected to debug.rttr.info:4123
Sending replay...
- File size: 23281071
- Compressing...
- Sending...
-> success
Will now send 6 stack frames

**Comment by Flamefire** [comment]
Ok so it is a crash. Loading the game shows some invalid state (NULL pointer in the roads of player 4)
Maybe the replay can show, why that happened. I'll look into it

**Comment by luk3ZPL** [comment]
Yes I have. I know that was strange. I've never seen that situation before.

http://www.mediafire.com/file/q2497ms26zaqofu/more_REPLAYS.7z/file

**Comment by luk3ZPL** [comment]
Ok, but now the question is why this road wasn't destroyed ?
Could it be (?):
- lag in later game
- async problems
- bug in code related to new "Trade addon" with ally player
I see something like this for the first time :P

**Comment by Flamefire** [comment]
It just wasn't done. All that was done was removing all player objects from enemy territory. A road is not an object and usually this is not a problem as a road is connected to a flag and gets removed with the flag. But in this case neither flag is removed due to the long road. I found and fixed this (see the commit) and it will be merged once all CI tests succeed.
